File: app/schemas/ingestion.py
from pydantic import BaseModel
from datetime import datetime
from typing import List, Optional

class IngestionJobResponse(BaseModel):
    id: str
    kb_id: str
    document_id: str
    task_id: Optional[str] = None
    status: str
    error_message: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

# 添加新的摄取启动响应模型
class IngestionStartResponse(BaseModel):
    """摄取任务启动响应"""
    id: str
    status: str
    message: str
    success: bool = True

class IngestionJobListResponse(BaseModel):
    jobs: List[IngestionJobResponse]
    total: int

class QueueStatsResponse(BaseModel):
    pending: int = 0
    running: int = 0
    total_tasks: int = 0
    completed: Optional[int] = 0
    failed: Optional[int] = 0
    timeout: Optional[int] = 0

class ChunkResponse(BaseModel):
    id: str
    kb_id: str
    document_id: str
    chunk_index: int
    content: str
    tags: List[str]
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/ingestion.ts
export interface IngestionJob {
  id: string;
  kb_id: string;
  document_id: string;
  status: IngestionJobStatus;
  created_at: string;
  updated_at: string;
  error_message?: string;
  progress?: number; // 添加进度字段
}

export enum IngestionJobStatus {
  PENDING = 'pending',
  RUNNING = 'running',
  COMPLETED = 'completed',
  FAILED = 'failed',
  CANCELLED = 'cancelled' // 添加取消状态
}

export interface IngestionStartRequest {
  document_id: string;
}

export interface IngestionStartResponse {
  id: string;
  status: IngestionJobStatus;
  message: string;
}

export interface BatchIngestionRequest {
  document_ids: string[];
}

export interface BatchIngestionResponse {
  jobs: IngestionStartResponse[];
  success_count: number;
  failed_count: number;
}

// 添加文档状态接口
export interface DocumentJobStatus {
  document_id: string;
  job_id?: string;
  status: IngestionJobStatus;
  progress?: number;
  error_message?: string;
}

export interface IngestionJobListResponse {
  jobs: IngestionJob[];
  total: number;
}